ZividOpenCV/CaptureUndistortRGB/CaptureUndistortRGB.cpp

```diff
@@ -25,32 +27,64 @@ int main()
-               << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
+        std::cout << "Enter <a><Return> for 2D only, and just <Return> for 3D" << std::endl;
+        const auto command = getInput();
+        bool use2D = false;
+        if(command == "a" || command == "A")
+        {
+            use2D = true;
+        }
+
+        cv::Mat rgb;
+
+        if(use2D)
+        {
+            std::cout << "2D only" << std::endl;
+
+            std::cout << "Configuring the camera settings" << std::endl;
+            auto settings = Zivid::Settings2D();
+            settings.set(Zivid::Settings2D::ExposureTime{ std::chrono::microseconds{ 10000 } });
+            settings.set(Zivid::Settings2D::Gain{ 1.0 });
+            settings.set(Zivid::Settings2D::Iris{ 35 });
+
+            std::cout << "Capture a 2D frame" << std::endl;
+            const auto frame = camera.capture2D(settings);
 
-        std::cout << "Capturing a frame" << std::endl;
-        const auto frame = camera.capture();
+            std::cout << "Get RGBA8 image from Frame2D" << std::endl;
+            const auto image = frame.image<Zivid::RGBA8>();
 
-        std::cout << "Setting up visualization" << std::endl;
-        Zivid::CloudVisualizer vis;
-        zivid.setDefaultComputeDevice(vis.computeDevice());
+            std::cout << "Converting RGBA8 image to OpenCV format" << std::endl;
+            rgb = imageToRGB(image);
+        }
+        else
+        {
+            std::cout << "Configuring the camera settings" << std::endl;
+            camera << Zivid::Settings::Iris{ 21 } << Zivid::Settings::ExposureTime{ std::chrono::microseconds{ 20000 } }
+                   << Zivid::Settings::Gain{ 1 } << Zivid::Settings::Brightness{ 1.0 };
 
-        std::cout << "Displaying the point cloud" << std::endl;
-        vis.showMaximized();
-        vis.show(frame);
-        vis.resetToFit();
+            std::cout << "Capturing a frame" << std::endl;
+            const auto frame = camera.capture();
 
-        std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
-        vis.run();
+            std::cout << "Setting up visualization" << std::endl;
+            Zivid::CloudVisualizer vis;
+            zivid.setDefaultComputeDevice(vis.computeDevice());
 
-        std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+            std::cout << "Displaying the point cloud" << std::endl;
+            vis.showMaximized();
+            vis.show(frame);
+            vis.resetToFit();
 
-        const auto pointCloud = frame.getPointCloud();
-        const auto rgb = pointCloudToRGB(pointCloud);
+            std::cout << "Running the visualizer. Blocking until the window closes" << std::endl;
+            vis.run();
+
+            std::cout << "Converting ZDF point cloud to OpenCV format" << std::endl;
+            const auto pointCloud = frame.getPointCloud();
+            rgb = pointCloudToRGB(pointCloud);
+        }
```

Comment:
`main()` has gotten fairly long now. Maybe you could refactor out some code here? For instance you could have one function `getImage2D()` and one `getImage3D()` (each extracted from the respective if and else branches here).

A nice side effect is then that you don't have to remove `const` from `rgb`.  You could  do `const atuo rgb = use2d  ? getImage2D() : getImage3d();` or something like that.